### PR TITLE
[AIRFLOW-XXX] Update airflow-jira release management script

### PR DESCRIPTION
**Pull request by ashb**
This PR is the changes I made to this script to make the last release --
it shows does a better job of finding merged PRs, linking "badly" tagged
PRs to jira tickets, and alos does a _rough_ pass at automating a
Changelog

### CODE REVIEW

**Comment by kaxil** [comment]
Can you update https://github.com/apache/airflow/blob/master/dev/README.md too, please?

**Comment by ashb** [comment]
You and your sensible solutions :P

**Review by kaxil — APPROVED** [review]

File: dev/README.md [hunk 1]
```diff
@@ -22,11 +22,38 @@
...
 
+- [Airflow Jira utility](#airflow-jira-utility)
 - [Airflow Pull Request Tool](#airflow-pull-request-tool)
 - [Airflow release signing tool](#airflow-release-signing-tool)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
+## Airflow Jira utility
+
+The `airflow-jira` script interact with the Airflow project in <https://issues.apache.org/jira/>. There are two modes of operation
```
Comment (XD-DENG) [inline dev/README.md:33]: "interact" -> "interacts"?

File: dev/README.md [hunk 2]
```diff
@@ -22,11 +22,38 @@
...
+
+
+- `compare` which examine issues in Jira based on the "Fix Version" field.
```
Comment (XD-DENG) [inline dev/README.md]: Change to
```
`compare` will examine...
```
or 
```
`compare` which examines...
```
?

File: dev/airflow-jira [hunk 3]
```diff
@@ -157,19 +159,27 @@ def cli():
...
-@click.argument('previous_version', default="")
+@click.option('--previous-version',
+              'previous_version',
+              help="Specify the previous tag on the working branch to limit"
+                   " searching for few commits to find the cherry-picked commits")
 @click.option('--unmerged', 'unmerged_only', help="Show unmerged issues only", is_flag=True)
 def compare(target_version, previous_version=None, unmerged_only=False):
     repo = git.Repo(".", search_parent_directories=True)
-    merges = get_merged_issues(repo, target_version, previous_version)
+    # Get  a list of issues/PRs that have been commited on the current branch.
```
Comment (XD-DENG) [inline dev/airflow-jira]: Seems duplicated spaces between `Get` and `a`

File: dev/airflow-jira [hunk 4]
```diff
@@ -195,19 +201,124 @@ def compare(target_version, previous_version=None, unmerged_only=False):
...
+
+    print("Commits on branch: {0:d}, {1:d} ({2}) yet to be cherry-picked".format(num_merged, sum(num_unmerged.values()), dict(num_unmerged)))
+
+
+@cli.command(short_help='Build a CHANGELOG grouped by Jira Issue type')
+@click.argument('previous_version')
+@click.argument('target_version')
+def changelog(previous_version, target_version):
+    repo = git.Repo(".", search_parent_directories=True)
+    # Get  a list of issues/PRs that have been commited on the current branch.
```
Comment (XD-DENG) [inline dev/airflow-jira]: Duplicated spaces here again.

**Review by XD-DENG — CHANGES_REQUESTED** [review]
A few very minor comments on typos.

Non-native English speaker so I put question marks at the end of my reviews ;-)

**Pull request merged**